Marlin/src/pins/stm32h7/pins_BTT_SKR_SE_BX_common.h

```diff
@@ -72,30 +84,44 @@
 //
 // Steppers
 //
-#define X_STEP_PIN                          PG13
+#define X_STEP_PIN                          PG13  // X
 #define X_DIR_PIN                           PG12
 #define X_ENABLE_PIN                        PG14
 #define X_CS_PIN                            PG10
 
-#define Y_STEP_PIN                          PB3
+#define Y_STEP_PIN                          PB3   // Y
 #define Y_DIR_PIN                           PD3
 #define Y_ENABLE_PIN                        PB4
 #define Y_CS_PIN                            PD4
 
-#define Z_STEP_PIN                          PD7
-#define Z_DIR_PIN                           PD6
-#define Z_ENABLE_PIN                        PG9
-#define Z_CS_PIN                            PD5
+#ifdef BX_SWAP_ZM_E1M
+  #define Z_STEP_PIN                        PA8   // E1
+  #define Z_DIR_PIN                         PC9
+  #define Z_ENABLE_PIN                      PD2
+  #define Z_CS_PIN                          PC8
+#else
+  #define Z_STEP_PIN                        PD7   // Z1
+  #define Z_DIR_PIN                         PD6
+  #define Z_ENABLE_PIN                      PG9
+  #define Z_CS_PIN                          PD5
+#endif
 
-#define E0_STEP_PIN                         PC14
+#define E0_STEP_PIN                         PC14  // E0
 #define E0_DIR_PIN                          PC13
 #define E0_ENABLE_PIN                       PC15
 #define E0_CS_PIN                           PI8
 
-#define E1_STEP_PIN                         PA8
-#define E1_DIR_PIN                          PC9
-#define E1_ENABLE_PIN                       PD2
-#define E1_CS_PIN                           PC8
+#ifdef BX_SWAP_ZM_E1M
```

Comment:
Is there any existing precedent for this type of swapping option? A quick search doesn't show anything.

I'd like more information in the PR description describing why this is needed.

Reply:
Oh, I see you mentioned that individual commits have more details...I'd rather see anything worth calling out pulled into the PR description, to make it easier to understand without having to dig around for it.

Reply:
I found one pre-existing example: `ARMED_SWAP_X_E1`

The syntax here isn't the same. To match what was done for ARMED it should just be `BX_SWAP_Z_E1`. Alternately ARMED could be changed to be consistent.

Let me know your thoughts.

Reply:
I mentioned the Chiron because it has `SWAP_Z_MOTORS`: https://github.com/MarlinFirmware/Marlin/blob/c31d9e717b207d1fa59ac942475ebcf15c1a3841/Marlin/src/pins/ramps/pins_TRIGORILLA_14.h#L148

There are likely others in the pins files, but there is no standard naming scheme.

I used `BX_SWAP_ZM_E1M` because that is how those connections are labeled on the motherboard.

Reply:
> I'd like more information in the PR description describing why this is needed.

I've copied out the commit descriptions into the PR description.

Reply:
Hmm...ok. This seems like a weak point for discoverability, without enough normalized naming convention to find them all.

But...it is consistent with the existing inconsistency, so I guess it's fine as-is.